internal/service/streamconnection/resource_schema.go

```
@@ -28,10 +29,26 @@ func ResourceSchema(ctx context.Context) schema.Schema {
 				},
 			},
 			"instance_name": schema.StringAttribute{
-				Required: true,
+				Optional: true,
 				PlanModifiers: []planmodifier.String{
 					stringplanmodifier.RequiresReplace(),
 				},
+				Validators: []validator.String{
+					stringvalidator.ConflictsWith(path.Expressions{
+						path.MatchRelative().AtParent().AtName("workspace_name"),
+					}...),
+				},
+			},
+			"workspace_name": schema.StringAttribute{
+				Optional: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
```

this means that if user changes workspace_name, Delete and Create will be called instead of Update.

qq: is this the behavior we want? there are 3 main alternatives:
- Destroy and create the resource again (current approach)
- Workspace_name can be handle in Update
- We want to raise an error that workspace_name can't be updated
-

---

Double checked and discussed with our team. This is intentional behavior. The same happens with the equivalent instance_name attribute today